pkg/splunk/enterprise/indexercluster.go

```diff
@@ -1069,3 +1074,62 @@ func RetrieveCMSpec(ctx context.Context, client splcommon.ControllerClient, cr *
 
 	return "", nil
 }
+
+// isIndexerClusterReadyForUpgrade checks if IndexerCluster can be upgraded if a version upgrade is in-progress
+// No-operation otherwise; returns bool, err accordingly
+func isIndexerClusterReadyForUpgrade(ctx context.Context, c splcommon.ControllerClient, cr *enterpriseApi.IndexerCluster) (bool, error) {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("isIndexerClusterReadyForUpgrade").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	eventPublisher, _ := newK8EventPublisher(c, cr)
+
+	// get the clusterManagerRef attached to the instance
+	clusterManagerRef := cr.Spec.ClusterManagerRef
+
+	// check if a search head cluster exists with the same ClusterManager instance attached
+	searchHeadClusterInstance := enterpriseApi.SearchHeadCluster{}
+	opts := []rclient.ListOption{
+		rclient.InNamespace(cr.GetNamespace()),
+	}
+	searchHeadList, err := getSearchHeadClusterList(ctx, c, cr, opts)
+	if err != nil {
+		if err.Error() == "NotFound" {
```

Comment:
can we use k8serror.IsNotFound(err) here, just donot want to assume the string will remain the same always.

Reply:
The method `getSearchHeadClusterList` uses the `client.List()` method that returns the `NotFound` error instead of the k8serror